price-feeder/cmd/price-feeder.go

```diff
@@ -82,21 +84,49 @@ func priceFeederCmdHandler(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
-	oracle := oracle.New()
-
 	ctx, cancel := context.WithCancel(context.Background())
 	g, ctx := errgroup.WithContext(ctx)
 
+	// listen for and trap any OS signal to gracefully shutdown and exit
+	trapSignal(cancel)
```

Comment:
curious, was there a reason to move this here?

Reply:
If it was below the `return fmt.Errorf` paths, it could've gone unused, which just seemed ugly

Reply:
Ok, I think this fine here too 👍